backend/services/deployments/server.go

```diff
@@ -221,5 +225,26 @@ func RunServer(ctx context.Context) error {
 		return http.ListenAndServeTLS(listen, cert, key, handler)
 	}
 
-	return http.ListenAndServe(listen, handler)
+	srv := &http.Server{
+		Addr:    listen,
+		Handler: handler,
+	}
+
+	go func() {
+		if err := srv.ListenAndServe(); err != nil && err != http.ErrServerClosed {
+			l.Fatalf("listen: %s\n", err)
+		}
+	}()
+
+	quit := make(chan os.Signal, 1)
+	signal.Notify(quit, unix.SIGINT, unix.SIGTERM)
+	<-quit
```

Comment:
If the go routine created here terminates unexpectedly, this process will continue to run doing nothing until a signal is received. We need some way of notifying the "parent" go routine that the "child" is stopping. You could for example achieve this by creating an additional `chan`nel to pass the error from the child and use a [select](https://go.dev/tour/concurrency/5) statement to listen on both `chan`nels.
```suggestion
	errChan := make(chan error, 1)
	go func() {
		if err := srv.ListenAndServe(); err != nil && err != http.ErrServerClosed {
			errChan <- err
		}
	}()

	quit := make(chan os.Signal, 1)
	signal.Notify(quit, unix.SIGINT, unix.SIGTERM)
	select {
	case sig := <-quit:
		l.Infof("received signal %s: terminating", sig)
	case err = <-errChan:
		l.Errorf("server terminated unexpectedly: %s", err.Error())
		return err
	}
```